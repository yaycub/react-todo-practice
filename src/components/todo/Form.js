import React, { useState } from 'react';
import { useTodos } from '../../hooks/useTodos';


const Form = () => {
  const [todo, setTodo] = useState('New Todo');

  const { handleSubmit } = useTodos(todo);

  const handleChange = ({ target }) => {
    setTodo(target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type='text' value={todo} onChange={handleChange} />
      <button>Create Task</button>
    </form>
  );
};

export default Form;
