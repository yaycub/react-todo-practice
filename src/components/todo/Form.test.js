import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form Component', () => {
  it('should render a form', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper).toMatchSnapshot();
  });
});
