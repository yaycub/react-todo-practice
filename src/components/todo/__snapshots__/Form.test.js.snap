// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render a form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="New Todo"
  />
  <button>
    Create Task
  </button>
</form>
`;
